src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -99,30 +101,36 @@ public void addMessage(List<Message> messages, Entity entity, String user) {
   }
 
   /**
-   * Gets all the messages by all the users
+   * Gets messages and cursor pointer information with size of PAGE_SIZE
+   * in the form of a QueryResult
    *
-   * @return a list of messages posted by all the users, or an empty list if
-   * no user has written a message. The List is sorted by time descending.
+   * @return a list of QueryResultList that store the data for user posts
+   *         and query cursor to the next page
    */
-  public List<Message> getAllMessages(){
-    List<Message> messages = new ArrayList<>();
+  public QueryResultList<Entity> getBatchMessages(HttpServletRequest req, HttpServletResponse resp)
+      throws IOException {
+    FetchOptions fetchOptions = FetchOptions.Builder.withLimit(PAGE_SIZE);
+
+    // If this servlet is passed a cursor parameter, let's use it.
+    String startCursor = req.getParameter("cursor");
```

Comment:
Since a bulk of the logic requires `request` and `response`, I decided to move most of the code to `MessageFeedServlet`. I think this logic applies to more of `MessageFeedServlet` and less of `Datastore`.